specs/bls_signature.md

```diff
@@ -130,17 +134,43 @@ g = Fq2([g_x, g_y])
 
 ### `bls_verify`
 
-Let `bls_verify(pubkey: Bytes48, message_hash: Bytes32, signature: Bytes96, domain: uint64) -> bool`:
+Let `bls_verify(pubkey: Bytes48, message: bytes, signature: Bytes96) -> bool`:
 
 * Verify that `pubkey` is a valid G1 point.
 * Verify that `signature` is a valid G2 point.
-* Verify that `e(pubkey, hash_to_G2(message_hash, domain)) == e(g, signature)`.
+* Verify that `e(pubkey, hash_to_G2(message)) == e(g, signature)`.
 
 ### `bls_verify_multiple`
 
-Let `bls_verify_multiple(pubkeys: List[Bytes48], message_hashes: List[Bytes32], signature: Bytes96, domain: uint64) -> bool`:
+Let `bls_verify_multiple(pubkeys: List[Bytes48], messages: List[bytes], signature: Bytes96) -> bool`:
 
 * Verify that each `pubkey` in `pubkeys` is a valid G1 point.
 * Verify that `signature` is a valid G2 point.
-* Verify that `len(pubkeys)` equals `len(message_hashes)` and denote the length `L`.
-* Verify that `e(pubkeys[0], hash_to_G2(message_hashes[0], domain)) * ... * e(pubkeys[L-1], hash_to_G2(message_hashes[L-1], domain)) == e(g, signature)`.
+* Verify that `len(pubkeys)` equals `len(messages)` and denote the length `L`.
+* Verify that `e(pubkeys[0], hash_to_G2(messages[0])) * ... * e(pubkeys[L-1], hash_to_G2(messages[L-1]) == e(g, signature)`.
+
+
+## Application layer
+
+In the specification documents, the following usage interface is followed:
+
+| `BLSDomain` | `Bytes8` | a BLS12-381 domain |
+| `BLSPubkey` | `Bytes48` | a BLS12-381 public key |
+| `BLSSignature` | `Bytes96` | a BLS12-381 signature |
+
+### `bls_verify`
+
+```python
+
+def bls_verify(pubkey: BLSPubkey, self_signed_object: Container, domain: BLSDomain) -> bool:
+    return bls.bls_verify(pubkey, domain + signed_root(self_signed_object), self_signed_object.signature)
```

Comment:
If we go this way (which I'm really not sold on!! I think there's a lot of value in reducing the depth of the abstraction tower that readers have to waddle through, and the `verify(pub, msg, sig)` format has a lot going for it given it's simple and familiar, and also symmetry with `bls_verify_multiple` is valuable), can we at least call it `bls_verify_signed_container`? Would make it clearer to readers what's going on imo.